wallet/dapps-communication.md

```diff
@@ -8,4 +8,67 @@ Initial meta issue: <https://github.com/massalabs/massa-standards/issues/13>
 
 ## Specification
 
+Here is pseudo code for the two points of view:
+
+- browser extension developer
+- dapp builder
+
+**As a browser extension developer, I would write...**
+
+browser extension code:
+
+```text
+// the extension register itself
+window.massaWalletProvider.addEventListener('loaded', () => {
```

Comment:
I'm not sure it will work like that.
The first thing to do I guess is to create an html element to listen event on.

Then, you can trigger the register event.

Reply:
I suggest something we have already done before in the massa-wallet with Antoine and it has worked really well (see here: https://github.com/massalabs/massa-standards/pull/26/files#diff-dad455c0814fa2229eaae244a61b26f2f312670c0d14b6da452cd46744739956R66-R93) and https://github.com/massalabs/massa-wallet/blob/main/inject.js

It is similar to what Nathan has, but it is a more advanced mechanism and is closer to what Metamask has done.

Reply:
I don't see why it is more advanced, can you clarify this part ?
From my point of view, using events on `window` element is a bad practice, creating potential conflict with other extension. I would prefer to use dedicated html elements.

Can you complete this section based on my remarks and what you suggest?

Reply:
@gregLibert  I dont understand what you mean by dedicated html elements ? An html element is a script, isn't it ? Since I dont understand your point here, maybe you can elaborate and expand your idea

Reply:
Yes, sorry.

Back when I was developing browser entensions more than 10 years ago, I used to do this in an html element (object under window.document), but it seems that now you can [create](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/EventTarget) an [EventTarget] (https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) wherever you want.

My point is that adding an event listner and firing an event directly on the window element is a bad practice  because if it is done correctly, there is no conflict, but if the page dev doesn't do it right, there can be side effects that are hard to debug.
I would prefer that we use a dedicated EventTarget. @Thykof suggests using the JS object `window.massaWalletProvider`, which is fine with me.

Reply:
More like this ?
```
// Wallet Side
new EventTarget().addEventListener('loaded', function(event) {
  console.log('Web3 Loaded', event.detail);
});

// Dapp Side
const myEvent = new CustomEvent('loaded', {
  detail: {
    message: 'You can register'
  }
});
myEventTarget.dispatchEvent(myEvent);
```

Reply:
so the html element to listen on in fact not a html element but an EventTarget. I have created one in the massa-library side that handle only registration of wallet providers.

Then each wallet provider (extension) define their own EventTarget